File: assignments/PPT/DSA/Assignments/Assignment_1/Q2.cpp
#include <stdio.h>
#include <iostream>
#include <vector>
#include <algorithm>
#include <unordered_map>
#include <iterator>

using namespace std;

/*

💡 Given an integer array nums and an integer val, remove all occurrences of val in nums in-place. The order of the elements may be changed. Then return the number of elements in nums which are not equal to val.

Consider the number of elements in nums which are not equal to val be k, to get accepted, you need to do the following things:

- Change the array nums such that the first k elements of nums contain the elements which are not equal to val. The remaining elements of nums are not important as well as the size of nums.
- Return k.

**Example :**
Input: nums = [3,2,2,3], val = 3
Output: 2, nums = [2,2,_*,_*]

**Explanation:** Your function should return k = 2, with the first two elements of nums being 2. It does not matter what you leave beyond the returned k (hence they are underscores)

*/


int solve(vector<int>& nums, int val) {
    int i = 0;
    int count = 0;
    int n = nums.size();
    for(int i = 0;i<n;i++){
        if(nums[i]!=val){
            nums[count] = nums[i];
            count++;
        }
    }
    return count;
}

int main(){
    // Remove the following three lines when running in terminal with input or some interface
    #ifndef ONLINE_JUDGE
        freopen("input2.txt","r",stdin);
    #endif

    int t = 1;
    cin>>t;
    while(t--){
        int n,target;

        cin>>n;

        vector<int> ar(n);

        for(int i = 0;i<n;i++){
            cin>>ar[i];
        }
        cin>>target;


        int ans = solve(ar, target);
        
        if(ans==0)
            cout<<"The array is empty!";
        else
            for(int i = 0;i<ans;i++){
                cout<<ar[i]<<" ";
            }
        cout<<endl;
    }

    return 0;
}

File: assignments/PPT/DSA/Assignments/Assignment_4/Q2.py
'''
Given two **0-indexed** integer arrays nums1 and nums2, return *a list* answer *of size* 2 *where:*

- answer[0] *is a list of all **distinct** integers in* nums1 *which are **not** present in* nums2*.*
- answer[1] *is a list of all **distinct** integers in* nums2 *which are **not** present in* nums1.

**Note** that the integers in the lists may be returned in **any** order.

**Example 1:**

**Input:** nums1 = [1,2,3], nums2 = [2,4,6]

**Output:** [[1,3],[4,6]]

**Explanation:**

For nums1, nums1[1] = 2 is present at index 0 of nums2, whereas nums1[0] = 1 and nums1[2] = 3 are not present in nums2. Therefore, answer[0] = [1,3].

For nums2, nums2[0] = 2 is present at index 1 of nums1, whereas nums2[1] = 4 and nums2[2] = 6 are not present in nums2. Therefore, answer[1] = [4,6].
'''

def solve(l1 : list[int], l2 : list[int]) -> list[list[int]]:
    rl = []
    rl.append(list({i for i in l1 if i not in l2}))
    rl.append(list({i for i in l2 if i not in l1}))
    return rl


if __name__ == "__main__":
    with open("./input2.txt",'r') as f:
        t = int(f.readline())
        for test_cases in range(t):
            l1 = [int(i) for i in f.readline().split(' ')]
            l2 = [int(i) for i in f.readline().split(' ')]
            print(solve(l1,l2))
  